browser/app/js/components/PolicyInput.js

```diff
@@ -1,98 +1,81 @@
-    }
-    
-    web.SetBucketPolicy({
+    web.SetBucketPolicy ({
       bucketName: this.props.currentBucket,
       prefix: this.prefix.value,
       policy: this.policy.value
     })
-      .then(() => {
-        dispatch(actions.setPolicies([{
-          policy, prefix
-        }, ...this.props.policies]))
-        this.prefix.value = ''
-      })
-      .catch(e => dispatch(actions.showAlert({
-        type: 'danger',
-        message: e.message,
-      })))
+        .then (() => {
+          dispatch (actions.setPolicies ([{
+            policy: this.policy.value,
+            prefix: this.prefix.value + '*',
+          }, ...this.props.policies]))
+          this.prefix.value = ''
+        })
+        .catch (e => dispatch (actions.showAlert ({
+          type: 'danger',
+          message: e.message,
+        })))
   }
 
-  render() {
+  render () {
     return (
-      <header className="pmb-list">
-        <div className="pmbl-item">
-          <input type="text"
-            ref={ prefix => this.prefix = prefix }
-            className="form-control"
-            placeholder="Prefix"
-            editable={ true } />
-        </div>
-        <div className="pmbl-item">
-          <select ref={ policy => this.policy = policy } className="form-control">
-            <option value={ READ_ONLY }>
-              Read Only
-            </option>
-            <option value={ WRITE_ONLY }>
-              Write Only
-            </option>
-            <option value={ READ_WRITE }>
-              Read and Write
-            </option>
-          </select>
-        </div>
-        <div className="pmbl-item">
-          <button className="btn btn-block btn-primary" onClick={ this.handlePolicySubmit.bind(this) }>
-            Add
-          </button>
-        </div>
-      </header>
+        <header className="policy__list">
+          <div className="policy__item">
+            <div className="form-group">
+              <input type="text"
+                     ref={ prefix => this.prefix = prefix }
+                     className="form-group__field form-group__field--sm"
+                     placeholder="Prefix"
+                     editable={ true }
+              />
+              <i className="form-group__bar" />
+            </div>
+          </div>
+          <div className="policy__item">
+            <select ref={ policy => this.policy = policy } className="form-group__field form-group__field--sm">
+              <option value={ READ_ONLY }>Read Only</option>
+              <option value={ WRITE_ONLY }>Write Only</option>
+              <option value={ READ_WRITE }>Read and Write</option>
+            </select>
+          </div>
+          <div className="policy__item">
+            <button className="btn btn--block btn--primary" onClick={ this.handlePolicySubmit.bind (this) }>
+              Add
+            </button>
+          </div>
+        </header>
     )
   }
 }
 
-export default connect(state => state)(PolicyInput)
```

Comment:
There may be an issue here with policyInput add/remove functionality. If Add button is clicked, default prefix with * is added to bucket and prevents adding any other prefix with Read only action. On the other hand, if the string  "*" is entered as prefix and Add button pressed,"*" is saved as bucket prefix. However, this also allows "someotherprefix" to also be added to policy. Likewise, if "prefix" is added first, followed by *, this allows both * and "prefix" as valid policy prefixes. The string "*" should be interpreted as default here.